projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-tool/tool.cpp

```diff
@@ -1040,6 +1042,102 @@ buffered_tracing_callback(rocprofiler_context_id_t /*context*/,
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind               = record->kind,
+                    .event_queue_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
+            else if(header->kind == ROCPROFILER_BUFFER_TRACING_KFD_EVENT_UNMAP_FROM_GPU)
+            {
+                auto* record = static_cast<rocprofiler_buffer_tracing_kfd_event_unmap_from_gpu_record_t*>(
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind                        = record->kind,
+                    .event_unmap_from_gpu_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
+            else if(header->kind == ROCPROFILER_BUFFER_TRACING_KFD_EVENT_DROPPED_EVENTS)
+            {
+                auto* record = static_cast<rocprofiler_buffer_tracing_kfd_event_dropped_events_record_t*>(
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind                        = record->kind,
+                    .event_dropped_events_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
+            else if(header->kind == ROCPROFILER_BUFFER_TRACING_KFD_PAGE_MIGRATE)
+            {
+                auto* record = static_cast<rocprofiler_buffer_tracing_kfd_page_migrate_record_t*>(
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind                = record->kind,
+                    .page_migrate_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
+            else if(header->kind == ROCPROFILER_BUFFER_TRACING_KFD_PAGE_FAULT)
+            {
+                auto* record = static_cast<rocprofiler_buffer_tracing_kfd_page_fault_record_t*>(
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind              = record->kind,
+                    .page_fault_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
+            else if(header->kind == ROCPROFILER_BUFFER_TRACING_KFD_QUEUE)
+            {
+                auto* record = static_cast<rocprofiler_buffer_tracing_kfd_queue_record_t*>(
+                    header->payload);
+
+                tool::tool_buffer_tracing_kfd_record_t rec = {
+                    .kind         = record->kind,
+                    .queue_record = *record
+                };
+
+                tool::write_ring_buffer(rec, domain_type::KFD);
+            }
```

Comment:
This is very repetitive. A combination of a lambda and std::variant might simplify things here